src/libraries/Common/tests/SingleFileTestRunner/SingleFileTestRunner.cs

```diff
@@ -27,6 +27,9 @@ public static int Main(string[] args)
         var asm = typeof(SingleFileTestRunner).Assembly;
         Console.WriteLine("Running assembly:" + asm.FullName);
 
+        // The current RemoteExecutor implementation is not compatible with the SingleFileTestRunner.
+        Environment.SetEnvironmentVariable("DOTNET_REMOTEEXECUTOR_SUPPORTED", "0");
```

Comment:
The reason why the existing users of the SingleFileTestRunner.cs didn't need this line is: https://github.com/dotnet/arcade/pull/6553

It is now redundant. Could you submit a removal of that in Arcade?

Reply:
By removal do you mean undoing dotnet/arcade#6553's changes, and then removing this line?

Reply:
The arcade change covers the existing use cases of the single file runner - the single file deployments (where assembly.location is null/empty, like the arcade PR checks).

With this PR, we'll be using the single file runner for non-single file scenarios as well. The line you're adding will disable remote executor for any consumer - single file or not.

Therefore, I think we won't need the arcade part anymore because remote executor will be disabled by this.

Reply:
Got it. Then, let me see if I understood correctly. In a sense, the arcade change and this line help with the same issue for this PR's purposes, so they are redundant in this case. Removing the arcade change while keeping this line will remove that redundancy without breaking other scenarios. Is this the appropriate course of action?

Reply:
Yep. Your change is making the arcade part redundant.